recipes-ni/cgroups-v2/nicreatecgroupsv2_1.0.bb

```diff
@@ -0,0 +1,79 @@
+SUMMARY = "NI cgroup v2 setup (mount/chown, cpusets, cpuacctgroups)"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+SRC_URI = "file://cgroupv2-common.sh \
+           file://createcpusets.sh \
+           file://createcpuacctgroups.sh \
+           file://nicreatecpusets \
+           file://nicreatecpuacctgroups \
+           file://nicreatecpusets.service \
+           file://nicreatecpuacctgroups.service \
+          "
+
+S = "${WORKDIR}"
+
+INHERIT += " systemd update-rc.d"
+
+SYSTEMD_AUTO_ENABLE = "enable"
+SYSTEMD_SERVICE:${PN} = "nicreatecpuacctgroups.service nicreatecpusets.service"
+
+INITSCRIPT_PACKAGES = "${PN}-cpusets ${PN}-cpuacct"
+INITSCRIPT_NAME:${PN}-cpusets = "nicreatecpusets"
+INITSCRIPT_PARAMS:${PN}-cpusets = "start 90 1 4 5 . stop 90 0 6 ."
+INITSCRIPT_NAME:${PN}-cpuacct = "nicreatecpuacctgroups"
+INITSCRIPT_PARAMS:${PN}-cpuacct = "start 91 2 4 5 . stop 91 0 6 ."
+
+FILES:${PN} += " /usr/lib/nicreatecgroupsv2/cgroupv2-common.sh \
+                  /usr/lib/nicreatecgroupsv2/createcpusets.sh \
+                  /usr/lib/nicreatecgroupsv2/createcpuacctgroups.sh \
+                "
+
+# systemd files
+FILES:${PN} += " /lib/systemd/system/nicreatecpuacctgroups.service \
+                  /lib/systemd/system/nicreatecpusets.service \
+                "
+
+# sysvinit files
+FILES:${PN} += " /etc/init.d/nicreatecpusets \
+                  /etc/init.d/nicreatecpuacctgroups \
+                "
+
+do_install() {
+    install -Dm755 ${WORKDIR}/cgroupv2-common.sh ${D}/usr/lib/nicreatecgroupsv2/cgroupv2-common.sh
+    install -Dm755 ${WORKDIR}/createcpusets.sh ${D}/usr/lib/nicreatecgroupsv2/createcpusets.sh
+    install -Dm755 ${WORKDIR}/createcpuacctgroups.sh ${D}/usr/lib/nicreatecgroupsv2/createcpuacctgroups.sh
+
+    if echo "${DISTRO_FEATURES}" | grep -q systemd; then
+        SYSTEMD_UNITDIR="${systemd_unitdir}"
+        install -Dm644 ${WORKDIR}/nicreatecpuacctgroups.service ${D}${SYSTEMD_UNITDIR}/system/nicreatecpuacctgroups.service
+        install -Dm644 ${WORKDIR}/nicreatecpusets.service ${D}${SYSTEMD_UNITDIR}/system/nicreatecpusets.service
+    else
+        install -Dm755 ${WORKDIR}/nicreatecpuacctgroups ${D}${sysconfdir}/init.d/nicreatecpuacctgroups
+        install -Dm755 ${WORKDIR}/nicreatecpusets ${D}${sysconfdir}/init.d/nicreatecpusets
+    fi
+}
+
+RDEPENDS_${PN} += "update-rc.d"
+
+pkg_postinst:${PN} () {
```

Comment:
Can you double check this postinst is actually needed? I believe the udate-rc.d class inherited above should handle this based on INITSCRIPT_* variables w/o requiring an explicit postinst.